app/__init__.py

```diff
@@ -1,11 +1,47 @@
-from flask import Flask
+import re
+
+from flask import Flask, request, abort, json
+from werkzeug.exceptions import HTTPException
+
+from app.helpers import make_error_msg
+from app.helpers.url import ALLOWED_DOMAINS_PATTERN
 from app.middleware import ReverseProxy
 
 # Standard Flask application initialisation
 
 app = Flask(__name__)
 app.wsgi_app = ReverseProxy(app.wsgi_app, script_name='/')
 
+
+# Add CORS Headers to all request
+@app.after_request
+def add_cors_header(response):
+    if (
+        'Origin' in request.headers and
+        re.match(ALLOWED_DOMAINS_PATTERN, request.headers['Origin'])
+    ):
+        response.headers['Access-Control-Allow-Origin'] = request.headers['Origin']
+        response.headers['Access-Control-Allow-Methods'] = 'GET'
+    return response
+
+
+# Reject request from non allowed origins
+@app.before_request
+def validate_origin():
+    if (
+        'Origin' in request.headers and
+        not re.match(ALLOWED_DOMAINS_PATTERN, request.headers['Origin'])
+    ):
+        abort(make_error_msg(403, 'Not allowed'))
```

Comment:
After discussion with @boecklic  we decided to always reject request without origin (see https://github.com/geoadmin/service-qrcode/pull/5/commits/4c7cd590c2731020f839df4d86b899eddb031ab2)

Reply:
this is implemented now